File: 백준/Silver/1874. 스택 수열/스택 수열.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    int n;
    cin >> n;
    
    stack<int> s;
    vector<string> operations;
    vector<int> sequence(n);
    
    for (int i = 0; i < n; i++) {
        cin >> sequence[i];
    }
    
    int current = 1;
    
    for (int i = 0; i < n; i++) {
        int target = sequence[i];
        
        while (current <= target) {
            s.push(current++);
            operations.push_back("+");
        }
        
        if (s.top() == target) {
            s.pop();
            operations.push_back("-");
        } else {
            cout << "NO" << '\n';
            return 0;
        }   
    }
    for (const string& op : operations) {
        cout << op << '\n';
    }
    return 0;
}

File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys
input = sys.stdin.readline

n = int(input())
sequence = [int(input()) for _ in range(n)]  # 입력 수열
stack = []
result = []
current = 1

for num in sequence:
    while current <= num:  # 목표 숫자에 도달할 때까지 push
        stack.append(current)
        result.append('+')
        current += 1
    
    if stack[-1] == num:  # 스택의 마지막 숫자가 목표 숫자와 같으면 pop
        stack.pop()
        result.append('-')
    else:
        print("NO")  # 목표 숫자를 만들 수 없으면 종료
        exit(0)

# 결과 출력
print('\n'.join(result))